### PR TITLE
Expose source.license through the API

### PR DESCRIPTION
**Pull request by jsoriano**
Add label indicating the distribution license of the package, as defined in https://github.com/elastic/package-spec/pull/355.

Required by Fleet for https://github.com/elastic/package-spec/issues/298 (https://github.com/elastic/package-spec/issues/298#issuecomment-1196023007).

### CODE REVIEW

File: packages/package.go [hunk 1]
```diff
@@ -121,6 +122,11 @@ type PolicyTemplate struct {
 	TemplatePath string `config:"template_path,omitempty" json:"template_path,omitempty" yaml:"template_path,omitempty"`
 }
 
+// Source contains metadata about the source of the package and its distribution.
+type Source struct {
+	License string `config:"license" json:"license"`
```
Comment (mtojek) [inline packages/package.go:127]: nit: omit empty and yaml?

**Review by mtojek — APPROVED** [review]

File: packages/package.go (hunk 1 above)
Comment (jsoriano) [reply packages/package.go]: Added

File: packages/package.go [hunk 2]
```diff
@@ -82,6 +82,7 @@ type BasePackage struct {
 	Title               *string              `config:"title,omitempty" json:"title,omitempty" yaml:"title,omitempty"`
 	Version             string               `config:"version" json:"version"`
 	Release             string               `config:"release,omitempty" json:"release,omitempty"`
+	Source              *Source              `config:"source,omitempty" json:"source,omitempty"`
```
Comment (andrewkroh) [inline packages/package.go:85]: Do you want a `yaml:"source,omitempty"` here too? This struct inconsistently uses `yaml` tags.

**Comment by jen-huang** [comment]
Thanks @jsoriano! I'll test the changes next week when I am back from PTO.

File: packages/package.go (hunk 2 above)
Comment (jsoriano) [reply packages/package.go]: Yeah, good point, we should take a look to these differences. @mtojek do you know why there are fields with different sets of annotations?

**Review by andrewkroh — APPROVED** [review]

**Comment by jsoriano** [comment]
Merging this so this is available soon.

@jen-huang let me know if this is not enough after your tests, but I think we cannot do something very different to expose this value :slightly_smiling_face:

**Review by mrodm — APPROVED** [review]
LGTM

**Comment by jen-huang** [comment]
> @jen-huang let me know if this is not enough after your tests, but I think we cannot do something very different to expose this value 🙂

I tested this for https://github.com/elastic/kibana/pull/137799 and things work fine 👍

Could I trouble you to also include a `LICENSE.txt` in one of these test packages? I need to test from the UI that we display a modal containing the license text.

**Comment by jsoriano** [comment]
> Could I trouble you to also include a `LICENSE.txt` in one of these test packages? I need to test from the UI that we display a modal containing the license text.

Do you need a real package with a `LICENSE.txt`? Or it is enough with a test package? For test package you can use the [`good` package](https://github.com/elastic/package-spec/tree/cf00245987ddfa9a9688e37294d23c3c268f5cc4/test/packages/good) in the package-spec test suite.
For real packages, we have to open the discussion about how to include the license text file in built packages, in the current integrations monorepo the license is in the root of the repository, and adding headers in files may be needed to properly apply the license.

**Comment by jen-huang** [comment]
No, I don't need a real package, just something to test my local EPR with. I'll try loading up the `good` package. Thanks!

**Comment by jsoriano** [comment]
Packages will eventually contain the license of the repository after this change https://github.com/elastic/elastic-package/pull/920

**Pull request merged**